### PR TITLE
configuration-aws-database covered by uptest

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

* Distilled SQLInstance abstraction out of platform-ref-aws
* configuration-aws-network as dependency
* uptest setup with waiting for --all configurations to avoid CRD establishing race conditions

I have:

- [ ] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

local uptest run
```
make e2e
...
00:44:58 [ OK ] running automated tests
```

plus uptest pipeline below

### CODE REVIEW

File: crossplane.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: meta.pkg.crossplane.io/v1alpha1
+kind: Configuration
+metadata:
+  name: platform-ref-aws
```
Comment (haarchri) [inline crossplane.yaml:4]: configuration-aws-database

File: examples/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: pkg.crossplane.io/v1
+kind: Configuration
+metadata:
+  name: platform-ref-aws
```
Comment (haarchri) [inline examples/configuration.yaml]: configuration-aws-database

File: examples/configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+spec:
+  package: xpkg.upbound.io/upbound/platform-ref-aws:v0.7.0
```
Comment (haarchri) [inline examples/configuration.yaml]: change to new repo url

**Review by haarchri — COMMENTED** [review]
@ytsarev thanks for the split - small nitpick

File: examples/configuration.yaml (hunk 3 above)
Comment (ytsarev) [reply examples/configuration.yaml]: @haarchri thanks a lot for the catch, updated!

**Comment by ytsarev** [comment]
@haarchri thanks a lot for review, sample Configuration updated, uptest is green, merging in :)

**Pull request merged**